lib/ansible/modules/identity/cyberark/cyberark_authentication.py

```diff
@@ -0,0 +1,324 @@
+            returned: success
+            type: bool
+            sample: true
+        use_shared_logon_authentication:
+            description: Whether or not Shared Logon Authentication will be used.
+            returned: success
+            type: bool
+            sample: true
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+import urllib
+import httplib
+import traceback
+import sys
+import json
+
+
+def processAuthentication(module):
+
+    # Getting parameters from module
+
+    api_base_url = module.params["api_base_url"]
+    validate_certs = module.params["validate_certs"]
+    username = module.params["username"]
+    password = module.params["password"]
+    new_password = module.params["new_password"]
+    use_shared_logon_authentication = module.params[
+        "use_shared_logon_authentication"]
+    use_radius_authentication = module.params["use_radius_authentication"]
+    state = module.params["state"]
+    cyberark_session = module.params["cyberark_session"]
+
+    # if in check mode it will not perform password changes
+    if module.check_mode and new_password is not None:
+        new_password = None
+
+    # Defining initial values for open_url call
+    headers = {'Content-Type': 'application/json'}
+    payload = ""
+
+    if state == "present": # Logon Action
+
+        # Different end_points based on the use of shared logon authentication
+        if use_shared_logon_authentication:
+
+            end_point = "/PasswordVault/WebServices/auth/Shared/RestfulAuthenticationService.svc/Logon"
+
+        else:
+
+            end_point = "/PasswordVault/WebServices/auth/Cyberark/CyberArkAuthenticationService.svc/Logon"
+
+            # The payload will contain username, password
+            # and optionally use_radius_authentication and new_password
+            payload_dict = {"username": username, "password": password}
+
+            if use_radius_authentication:
+                payload_dict["useRadiusAuthentication"] = use_radius_authentication
+
+            if new_password is not None:
+                payload_dict["newPassword"] = new_password
+
+            payload = json.dumps(payload_dict)
+
+    else: # Logoff Action
+
+        # Get values from cyberark_session already established
+        api_base_url = cyberark_session["api_base_url"]
+        validate_certs = cyberark_session["validate_certs"]
+        use_shared_logon_authentication = cyberark_session[
+            "use_shared_logon_authentication"]
+        headers["Authorization"] = cyberark_session["token"]
+
+        # Different end_points based on the use of shared logon authentication
+        if use_shared_logon_authentication:
+            end_point = "/PasswordVault/WebServices/auth/Shared/RestfulAuthenticationService.svc/Logoff"
+        else:
+            end_point = "/PasswordVault/WebServices/auth/Cyberark/CyberArkAuthenticationService.svc/Logoff"
+
+    try:
```

Comment:
having such a large try/except can lead to troubleshooting problems for the end user. Please break it up into smaller blocks.